packages/core-components/src/lib/unsorted/viz/table/GroupRow.svelte

```diff
@@ -0,0 +1,332 @@
+<script>
+	import { createEventDispatcher } from 'svelte';
+	import TableChevronToggle from './TableChevronToggle.svelte';
+	import Delta from './Delta.svelte';
+	import { safeExtractColumn } from './datatable.js';
+	import {
+		formatValue,
+		getFormatObjectFromString
+	} from '@evidence-dev/component-utilities/formatting';
+	import { getContext } from 'svelte';
+	import { propKey } from '@evidence-dev/component-utilities/chartContext';
+	$: props = getContext(propKey);
+
+	export let groupName;
+	export let currentGroup; // summarizedGroups[groupName]
+	export let toggled;
+	export let columnSummary;
+	export let rowNumbers;
+	export let backgroundColor = undefined;
+
+	const dispatch = createEventDispatcher();
+
+	function toggleGroup() {
+		dispatch('toggle', { groupName });
+	}
+</script>
+
+<tr
+	on:click={toggleGroup}
+	class="cursor-pointer hover:bg-gray-100 w-full border-t-[1px] border-gray-200"
+	role="button"
+	tabindex="0"
+	on:keypress={(e) => e.key === 'Enter' && toggleGroup()}
+	style:background-color={backgroundColor}
+>
+	<!-- {#if rowNumbers}
+<td class="index w-[2%]" />
+{/if} -->
+
+	{#each $props.columns.length > 0 ? $props.columns : columnSummary.filter((d) => d.show === true) as column, j}
+		{@const format = column.totalFmt
+			? getFormatObjectFromString(column.totalFmt)
+			: column.fmt
+			? getFormatObjectFromString(column.fmt)
+			: columnSummary.format}
+		{@const useCol = safeExtractColumn(column, columnSummary)}
+		{@const column_format = column.fmt
+			? getFormatObjectFromString(column.fmt, useCol.format?.valueType)
+			: useCol.format}
+		{#if j === 0}
+			<td class="font-medium" colspan={rowNumbers ? 2 : 1} style="padding: 3px;">
+				<div class="items-center gap-2 align-top">
+					<span class="inline-flex print-hidden chev"><TableChevronToggle {toggled} /></span>
+					{groupName}
+				</div>
+			</td>
+		{:else}
+			<td class="{columnSummary[j].type} font-medium">
+				{#if column.contentType === 'delta'}
+					<Delta
+						value={currentGroup[column.id]}
+						downIsGood={column.downIsGood}
+						format={column_format}
+						columnUnitSummary={useCol.columnUnitSummary}
+						showValue={column.showValue}
+						deltaSymbol={column.deltaSymbol}
+						align={column.align}
+					/>
+				{:else}
+					{formatValue(currentGroup[column.id], format, useCol.columnUnitSummary)}
+				{/if}
+			</td>
+		{/if}
+	{/each}
+</tr>
+
+<style>
```

Comment:
Are all these classes used?

Reply:
We should also use tailwind instead of direct styles here